internal/core/src/segcore/FieldIndexing.cpp

```diff
@@ -45,39 +49,140 @@ VectorFieldIndexing::BuildIndexRange(int64_t ack_beg,
     }
 }
 
+void
+VectorFieldIndexing::GetDataFromIndex(const int64_t* seg_offsets,
+                                int64_t count,
+                                int64_t element_size,
+                                void* output) {
+    auto ids_ds = std::make_shared<knowhere::DataSet>();
+    ids_ds->SetRows(count);
+    ids_ds->SetDim(1);
+    ids_ds->SetIds(seg_offsets);
+    ids_ds->SetIsOwner(false);
+
+    auto& vector = index_->GetVector(ids_ds);
+
+    std::memcpy(output, vector.data(), count * element_size);
```

Comment:
This can't work with string data

Reply:
ok, in this case, variable vector can be inferred as uint8_t vector.